src/cache-utils.ts

```diff
@@ -34,13 +34,29 @@ export const getPackageManagerInfo = async (packageManager: string) => {
 export const getCacheDirectoryPath = async (
   packageManagerInfo: PackageManagerInfo
 ) => {
-  const pathList = await Promise.all(
+  const pathOutputs = await Promise.allSettled(
     packageManagerInfo.cacheFolderCommandList.map(command =>
       getCommandOutput(command)
     )
   );
 
-  const cachePaths = pathList.filter(item => item);
+  pathOutputs
+    .filter(output => output.status === 'rejected')
+    .forEach(output =>
+      core.warning(
+        `getting cache directory path failed: ${
+          (output as PromiseRejectedResult).reason
+        }`
+      )
+    );
+
+  const cachePaths = pathOutputs
+    .filter(
+      output =>
+        output.status === 'fulfilled' &&
+        (output as PromiseFulfilledResult<string>).value
+    )
+    .map(output => (output as PromiseFulfilledResult<string>).value);
```

Comment:
Could you please change it to this:
```ts
  const results = pathOutputs.map(item => {
    if (item.status === 'fulfilled') {
      return item.value;
    } else {
      core.info(`[warning]${item.reason}`);
    }

    return '';
  });

  const cachePaths = results.filter(item => item);
```

Reply:
I agree, but I would update the warning message to be more clear
```js
core.info(`[warning]getting cache directory path failed: ${item.reason}`);
```